### PR TITLE
Refactor hostname validation to use Zod schema (HMS-11227)

### PR DESCRIPTION
**Pull request by kingsleyzissou**
## Summary

Refactor hostname validation to use Zod schema for runtime validation, moving validator logic from component utilities to the Redux store for better reusability and maintainability. This centralizes schema definitions and ensures consistent validation across the application.

## Changes

- Add Zod schema for hostname validation in `src/store/slices/wizard/system/schemas.ts`
- Create dedicated validator module in `src/store/slices/wizard/system/validators.ts` with hostname validation function
- Migrate `useValidation` hook to use the new `validateHostname` function instead of inline validation logic
- Update Hostname component tests to verify validation through the new validator
- Add comprehensive test coverage for hostname validation edge cases in store slice tests
- Add Zod as a runtime dependency for schema validation

### CODE REVIEW

File: src/store/slices/wizard/system/schemas.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import z from 'zod';
+
+export const hostnameSchema = z
+  .string()
+  .max(64, 'Hostname must be no more than 64 characters.')
+  // TODO: Consider splitting this complex regex into multiple Zod .regex() / .refine()
+  // checks to provide more specific, user-friendly error messages (e.g., "cannot contain
+  // consecutive dots", "cannot start with hyphen", etc.)
```
Comment (regexowl) [inline src/store/slices/wizard/system/schemas.ts:8]: Uuh, this is cool!

File: src/store/slices/wizard/system/validators.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  if (!hostname) return {};
+
+  const result = hostnameSchema.safeParse(hostname);
+  if (result.success) {
+    return {};
+  }
+
+  return {
+    // NOTE: just return the first issue for now, this
+    // is all our frontend currently support
```
Comment (regexowl) [inline src/store/slices/wizard/system/validators.ts:15]: Yup, sounds good. We can iterate on this.

**Review by regexowl — APPROVED** [review]
Ok, looking at the implementation - all for this! The one added dep is worth it ❤️

Also love the validation specific unit tests. There is a slight overlap with already existing unit tests - we could clean those up and have smaller files focused on one concern.

**Comment by kingsleyzissou** [comment]
> Also love the validation specific unit tests. There is a slight overlap with already existing unit tests - we could clean those up and have smaller files focused on one concern.

Ya absolutely, since I was refactoring, I really wanted to make sure I wasn't breaking existing behaviour.
I will do a clean up at the end
